library/core/src/slice/iter.rs

```diff
@@ -710,18 +710,17 @@ where
             return None;
         }
 
-        let idx_opt = {
-            // work around borrowck limitations
-            let pred = &mut self.pred;
-            self.v.iter().position(|x| (*pred)(x))
-        };
-        match idx_opt {
+        match self.v.iter().position(|x| (self.pred)(x)) {
             None => self.finish(),
             Some(idx) => {
                 let tmp = mem::replace(&mut self.v, &mut []);
-                let (head, tail) = tmp.split_at_mut(idx);
-                self.v = &mut tail[1..];
-                Some(head)
+                // LLVM does not manage to remove the bounds check in this implementation:
+                // let (head, tail) = tmp.split_at_mut(idx);
+                // self.v = &mut tail[1..];
+                // Some(head)
```

Comment:
I don't think this is a particularly useful comment. However, I'd love to see a comment here saying something like "Return [..idx], drop idx, and leave [idx + 1..] for the next iteration". It took me a bit of thinking to figure out that's what's being done here; I don't think it's immediately obvious.